src/presentation/wallet/transactions/index.tsx

```diff
@@ -59,22 +63,46 @@ const TransactionsView: React.FC<TransactionsProps> = ({
   // TxDetails Modal
   const [modalTxDetails, setModalTxDetails] = useState<TxDisplayInterface>();
 
+  // submarine swap bottom sheet modal
+  const [showBottomSheet, setShowBottomSheet] = useState(false);
+  const [action, setAction] = useState('');
+
   // Save mnemonic modal
   const [isSaveMnemonicModalOpen, showSaveMnemonicModal] = useState(false);
   const handleSaveMnemonicClose = () => showSaveMnemonicModal(false);
   const handleSaveMnemonicConfirm = async () => {
     await browser.tabs.create({ url: `home.html#${BACKUP_UNLOCK_ROUTE}` });
   };
+
   const handleReceive = () => {
     if (!isWalletVerified) {
       showSaveMnemonicModal(true);
+      return;
+    }
+    setAction('receive');
+    if (state.canSubmarineSwap) {
+      setShowBottomSheet(true);
     } else {
       history.push(`${RECEIVE_ADDRESS_ROUTE}/${state.assetHash}`);
     }
   };
   const handleSend = async () => {
+    setAction('send');
     await dispatch(setAsset(state.assetHash));
-    history.push(SEND_ADDRESS_AMOUNT_ROUTE);
+    if (state.canSubmarineSwap) {
+      setShowBottomSheet(true);
+    } else {
+      history.push(SEND_ADDRESS_AMOUNT_ROUTE);
+    }
+  };
+  const handleLightningSelection = (lightning: boolean) => {
+    if (lightning) {
+      if (action === 'send') history.push(LIGHTNING_ENTER_INVOICE_ROUTE);
+      if (action === 'receive') history.push(LIGHTNING_ENTER_AMOUNT_ROUTE);
+    } else {
+      if (action === 'send') history.push(SEND_ADDRESS_AMOUNT_ROUTE);
+      if (action === 'receive') history.push(`${RECEIVE_ADDRESS_ROUTE}/${state.assetHash}`);
+    }
```

Comment:
I am more fan of one single method per action instead of a boolean with brancing condition:

you can split in `handleLightningSelection` and `handleLiquidSelection` without boolean param for both since implicit

Reply:
You're right, much cleaner with 97b0924